unstructured/partition/lang.py

```diff
@@ -201,3 +202,56 @@ def _get_all_tesseract_langcodes_with_prefix(prefix: str):
     Get all matching tesseract langcodes with this prefix (may be one or multiple variants).
     """
     return [langcode for langcode in PYTESSERACT_LANGS if langcode.startswith(prefix)]
+
+
+def _convert_to_standard_langcode(lang: str) -> str:
+    """
+    Convert a language code to the standard internal language code format.
+    """
+    # convert to standard ISO 639-3 language code
+    lang_iso639 = iso639.Language.match(lang[:3].lower())
+    return lang_iso639.part3
+
+
+def detect_languages(
+    text: str,
+    languages: List[str] = ["auto"],
+) -> List[str]:
+    """
+    Detects the list of languages present in the text (in the default "auto" mode),
+    or formats and passes through the user inputted document languages if provided.
+    """
+    if text.strip() == "":
+        return ["eng"]  # english as default
+
+    DetectorFactory.seed = 0
```

Comment:
potentially a comment around this - if I were looking at this the first time it's not clear what this is for